numpy/lib/tests/test_function_base.py

```diff
@@ -167,6 +167,29 @@ def test_returned(self):
         avg, scl = average(y, weights=w2, axis=1, returned=True)
         assert_array_equal(scl, np.array([1., 6.]))
 
+    def test_subclasses(self):
+        class subclass(np.ndarray):
+            pass
+        a = np.array([[1,2],[3,4]]).view(subclass)
+        w = np.array([[1,2],[3,4]]).view(subclass)
+
+        assert_equal(type(np.average(a, weights=w)), subclass)
```

Comment:
Should that not be better as `assert_(type(np.average(a, weights=w)) is subclass)`?